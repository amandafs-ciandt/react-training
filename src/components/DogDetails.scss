.dog-details {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 1rem;
  border-radius: 1rem;

  &__main-info {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  &__image {
    width: 8rem;
    height: 8rem;
    position: relative;
    overflow: hidden;
    border-radius: 50%;
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center;
  }

  &__info {
    margin-left: 3rem;
  }

  &__actions {
    display: flex;
    flex-direction: column;

    button:first-child {
      margin-bottom: 1rem;
    }
  }

  h2,
  p {
    margin: 0;
  }
}
