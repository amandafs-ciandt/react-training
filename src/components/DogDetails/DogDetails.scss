.dog-details {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  padding: 1rem;
  text-align: center;

  img {
    width: 15rem;
  }

  .MuiTypography-root {
    margin: 0;
  }

  .MuiCardActions-root {
    justify-content: center;
  }
}
