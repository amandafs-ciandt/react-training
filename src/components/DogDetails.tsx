import React from 'react';
import { useStore } from 'effector-react';
import _ from 'lodash';

import { selectedDog } from '../effector/store';

import './DogDetails.scss';
import { Button, notification } from 'antd';
import { scoldDog } from '../effector/event';

const DogDetails = () => {
  const dog = useStore(selectedDog);

  const bark = () => {
    notification.open({
      message: 'Woof Woof!',
    });
  };

  return (
    <div className='dog-details'>
      <div className='dog-details__main-info'>
        <div
          className='dog-details__image'
          style={{ backgroundImage: `url(${dog.image})` }}></div>
        <h2>{_.upperFirst(dog.breed)}</h2>
      </div>
      <div className='dog-details__info'>
        <div className='dog-details__actions'>
          <Button onClick={() => scoldDog()}>Scold!</Button>
          <Button onClick={bark}>Bark!</Button>
        </div>
      </div>
    </div>
  );
};

export default DogDetails;
