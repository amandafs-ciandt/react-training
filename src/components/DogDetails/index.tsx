import React from "react";
import { useStore } from "effector-react";
import _ from "lodash";

import { selectedDog } from "../../effector/store";
import { scoldDog } from "../../effector/event";

import "./DogDetails.scss";
import { notification, Card, Avatar, Typography } from "antd";

const { Title } = Typography;

const DogDetails = () => {
  const dog = useStore(selectedDog);

  const bark = () => {
    notification.open({
      message: "Woof Woof!",
    });
  };

  const avatarSize = 150;
  const titleSize = 4;

  return (
    <div className="dog-details">
      <Card
        style={{ width: 300 }}
        cover={<Avatar size={avatarSize} src={dog.image} />}
        actions={[
          <p className="scold-action" onClick={() => scoldDog()}>
            Scold!
          </p>,
          <p className="bark-action" onClick={bark}>
            Bark!
          </p>,
        ]}
      >
        <Title level={titleSize}>{_.upperFirst(dog.breed)}</Title>
      </Card>
    </div>
  );
};

export default DogDetails;
