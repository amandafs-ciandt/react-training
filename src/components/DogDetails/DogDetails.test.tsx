import React from "react";
import * as effector from "effector-react";
import { shallow, ShallowWrapper, mount, ReactWrapper } from "enzyme";

import DogDetails from ".";
import { Avatar, notification } from "antd";
import { ActionButtonProps } from "antd/lib/modal";

const setUp = () => {
  return shallow(<DogDetails />);
};

const selectedDog = {
  breed: "bulldog",
  image: "https://images.dog.ceo/breeds/bulldog-french/n02108915_4474.jpg",
  scolded: 3,
};

const spy = jest.spyOn(effector, "useStore");
spy.mockReturnValue(selectedDog);

describe("DogDetails", () => {
  let dogDetailsComponent: ShallowWrapper | ReactWrapper;

  beforeEach(() => {
    dogDetailsComponent = setUp();
  });

  it("should work", () => {
    expect(dogDetailsComponent).toMatchSnapshot();
  });

  it("should display the dog details card correctly", () => {
    const cardAvatar = dogDetailsComponent.find("Card").prop("cover") as Avatar;
    const cardTitle = dogDetailsComponent.find("Title").props().children;
    const cardButtons = dogDetailsComponent
      .find("Card")
      .prop("actions") as ActionButtonProps;

    expect(cardAvatar.props.src).toEqual(selectedDog.image);
    expect(cardTitle).toEqual("Bulldog");
    expect(cardButtons[0].props.children).toEqual("Scold!");
    expect(cardButtons[1].props.children).toEqual("Bark!");
  });

  it("should show notification when 'bark!' is clicked", () => {
    const barkSpy = jest.spyOn(notification, "open");
    dogDetailsComponent = mount(<DogDetails />);

    const barkButton = dogDetailsComponent.find(".bark-action");

    barkButton.simulate("click");

    expect(barkSpy).toHaveBeenCalledWith({
      message: "Woof Woof!",
    });
  });

  /* it("should call scoldDog event 'scold!' is clicked", () => {
    const scoldSpy = jest.spyOn(notification, "open");
    dogDetailsComponent = mount(<DogDetails />);

    const scoldButton = dogDetailsComponent.find(".scold-action");

    scoldButton.simulate("click");

    expect(scoldSpy).toHaveBeenCalled();
  }); */
});
