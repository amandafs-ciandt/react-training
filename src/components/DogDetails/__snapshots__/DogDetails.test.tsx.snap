// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogDetails should work 1`] = `ShallowWrapper {}`;
